src/stackmap_checker/stmap.c

```diff
@@ -78,37 +78,32 @@ void assert_valid_reg_num(unw_regnum_t reg)
     }
 }
 
-uint64_t stmap_get_location_value(stack_map_t *sm, location_t loc,
-        uint64_t *regs, void *frame_addr)
+void stmap_get_location_value(stack_map_t *sm, location_t loc,
+        uint64_t *regs, void *frame_addr, void **loc_value, uint64_t loc_size)
 {
-    uint64_t *dest_addr;
-    uint64_t addr, value;
+    uint64_t addr = 0;
+    *loc_value = malloc(loc_size);
     switch (loc.kind) {
         case REGISTER:
             assert_valid_reg_num(loc.dwarf_reg_num);
-            return regs[loc.dwarf_reg_num];
+            memcpy(*loc_value, &regs[loc.dwarf_reg_num], loc_size);
+            break;
         case DIRECT:
-            dest_addr = (uint64_t *)malloc(sizeof(uint64_t));
             addr = (uint64_t)frame_addr + loc.offset;
-            memcpy(dest_addr, (void *)addr, loc.size);
-            value = *dest_addr;
-            free(dest_addr);
-            return value;
+            memcpy(*loc_value, (void *)addr, loc_size);
+            break;
         case INDIRECT:
-            dest_addr = (uint64_t *)malloc(sizeof(uint64_t));
-            addr = (uint64_t)frame_addr + loc.offset;
-            memcpy(dest_addr, (void *)*(uint64_t *)addr, loc.size);
-            value = *dest_addr;
-            free(dest_addr);
-            return value;
+            errx(1, "Not implemented.");
```

Comment:
So this is a regression?

Reply:
Not quite, since this was never actually properly implemented/tested. I am going to implement this in a different branch. I am still trying to work out how to trigger register spills (indirect locations handle register spills).